source/_integrations/landisgyr_heat_meter.markdown

```diff
@@ -0,0 +1,88 @@
+---
+title: Landis+Gyr Heat Meter
+description: Instructions on how to integrate your Landis+Gyr Heat Meter device into Home Assistant.
+ha_category:
+  - Energy
+  - Sensor
+ha_iot_class: Local Polling
+ha_config_flow: true
+ha_release: 2022.7
+ha_domain: landisgyr_heat_meter
+ha_codeowners:
+  - '@vpathuis'
+ha_platforms:
+  - sensor
+ha_integration_type: integration
+---
+
+The Landis+Gyr Heat Meter integration for Home Assistant allows you to read the meter values from Ultraheat devices by Landis & Gyr. These devices are commonly used for district heating. The integration has been tested with the Landis & Gyr Ultraheat 50 (UH50) type LUGCUH50. Other models are likely to work as well.
+
+The device is read through the optical interface. An (USB) IR reader is needed and connected to Home Assistant.
+
+<img src='/images/integrations/landisgyr_heat_meter/usb_ir_reader.png' />
+
+<div class='note warning'>
+Every time the Heat Meter values are read, battery time of the device will supposedly go down by about 30 minutes. To prevent battery drain, the integration will not poll automatically.
+</div>
+
+{% include integrations/config_flow.md %}
+
+## Sensors
+
+The integration will create the following sensors:
+
+- heat usage (MWh)
+- volume usage (m3)
+
+To be compatible with the Home Assistant energy units of measurement, heat usage is converted to MWh, from usage in GJ which is supplied by the device, using conversion factor: 1 GJ = 0.277778 MWh.  
+
+Further data that is read from the device is added as diagnostic entities: 
+
+- heat usage measured in GJ, as is read from the device before conversion
+- ownership number
+- volume previous year (m3)
+- heat previous year (MWh)
+- error number
+- device number
+- measurement period minutes
+- power max kw
+- power max previous year kw
+- flow rate max m3ph
+- flow rate max previous year m3ph
+- flow temperature max c
+- return temperature max c
+- flow temperature max previous year c
+- return temperature max previous year c
+- operating hours
+- fault hours
+- fault hours previous year
+- yearly set day
+- monthly set day
+- meter date time
+- measuring range m3ph
+- settings and firmware
+- flow hours
+
+## Energy Dashboard
+
+Either heat usage or volume usage can be used as "Gas" on the energy dashboard. If you want to supply a price per MWh, make sure to apply the conversion factor first.
+
+## Polling the device
+
+Polling is not automatic (see battery warning). This means that after adding the integration all values will initially be unknown.
+To read the values from the device, create an automation that will update one of the entities. The other entities will be updated as well.
```

Comment:
That sounds off? We sound not make a broken sensor out of the box, not should we ever allow sensors that require manually created automations to function.

I would suggest restoring the state of this sensor (so we don't need to poll on startup) and poll, e.g., once a week or something like that.

Reply:
Agreed. This text was a leftover from an earlier version. State is now indeed restored on startup. It's also polled after adding the integration so the user should not see unknown values.

Reply:
Is that acceptable? I'd rather leave the polling to the user, but if it's not compliant with HA standards, of course I will change this.

Reply:
Well it doesn't violate any standards I think, but no everybody knows or uses YAML, confronting them with a YAML blob to get a working sensor, IMHO isn't the user experience we should or want to provide.

> State is now indeed restored on startup. It's also polled after adding the integration so the user should not see unknown values.

That means it updates once and never again? That is also weird. So it worked, but doesn't anymore (from a user perspective).

Alternatively, poll daily or weekly, and keep the entity disabled by default (and don't poll when it is). This allows the user to enable it if they want.

If they want another polling schedule, they can still opt to do that (e.g., do it weekly, monthly, or every minute).

Reply:
Thanks for the feedback. Disabling entities won't be necessary I think, because that would just disable all functionality and in that case the user could just remove the integration. 

Creating the automation would not require using YAML. The example is just shownn for convenience. I now understand it has the oposite effect :-o.

The idea behind the chosen solution is:
- the user probably only wants to poll daily as a maximum, due to the risk of battery drain (this could only be replaced by a specialist, so expensive). This is ultimately for the user to decide. 
- the user wants as much control on getting the most relevant detail as possible, within this limitation, so probably polling late in the day, or week, or perhaps before and after the weekend. Allowing the user to do it manually will give max control. 

I also looked at options to set a time in config flow, but I felt that was not the right place. 

What I could do is add a setting in the config flow to enable or disable polling? If enabled, it will poll once weekly (without time option). If disabled, the user can do it manually using automations. How does that sound? 

This will require changes to the PR that is already merged. How does that work, should I just add another PR?

Reply:
> Creating the automation would not require using YAML. The example is just shownn for convenience. I now understand it has the oposite effect :-o.

Things remain the same, if it is a polling integration, we need to provide a sane default and usable polling frequency. If one decides to change, that is up to them.

> What I could do is add a setting in the config flow to enable or disable polling?

That already exists. For every integration.

Reply:
> Things remain the same, if it is a polling integration, we need to provide a sane default and usable polling frequency. If one decides to change, that is up to them.

Alright, understood.
 
> That already exists. For every integration.
Oh yes I see now. That's very convenient! :)
I'll apply the change and then also change the documentation.

Reply:
I've submitted the change here: https://github.com/home-assistant/core/pull/77078
I've also changed the documentation accordingly. 
I look forward to your further suggestions or feedback!